platform/default/src/mbgl/gfx/headless_frontend.cpp

```diff
@@ -140,28 +140,26 @@ PremultipliedImage HeadlessFrontend::readStillImage() {
     return backend->readStillImage();
 }
 
-HeadlessFrontend::RenderResult HeadlessFrontend::render(Map& map) {
+HeadlessFrontend::RenderResult HeadlessFrontend::render(Map& map,
+                                                        std::exception_ptr *error) {
     HeadlessFrontend::RenderResult result;
-    std::exception_ptr error;
     gfx::BackendScope guard { *getBackend() };
 
     map.renderStill([&](const std::exception_ptr& e) {
         if (e) {
```

Comment:
What kind of exceptions can `std::exception_ptr` contain? Can all be safely ignored?

Reply:
We ignore the resource error like the font file download failed, you know, HeadlessFrontend::render should handle this kind of error, do you read the unit test it just simulates this condition, the unit test `TEST(Map, ResourceError)` try to take a snapshot, but the font resource download failed at first, and success in the second try.  Before this edition, the snapshot can process success because of the render function abort by resource error(break the render loop and throw the error), and after this edition the snap shot could success.
error not be ignored, SDK(Adr, iOS, etc.) also can capture it via parameters.